### PR TITLE
Added principalBreakingTerm

### PR DESCRIPTION
**Pull request by avantassel**
Added principalBreakingTerm to know when principals payments will be more than interest payments

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -22,8 +22,9 @@ var amortizationCalc = function(amount, rate, totalTerm, amortizeTerm, principal
       monthlyIntPaid,
       monthlyPrincPaid,
       summedAmortize = {},
-      montlyPrincipalPayment = 0,
-      principalPayment = (!!principalPayment) ? parseInt(principalPayment,10) : 0;
+      principalPayment = (!!principalPayment) ? parseInt(principalPayment,10) : 0,
+      principalBreakingTerm = 0,
+      montlyPrincipalPayment = 0;
```
Comment (ascott1) [inline index.js]: It looks like this is a typo and this should be `monthlyPrincipalPayment`

File: test/test.js [hunk 2]
```diff
@@ -75,6 +75,11 @@ exports['After 30 years a borrower with a 30 year, $180,000 loan with a 4.25% in
   test.done();
 };
 
+exports['After 30 years a borrower with a 30 year, $180,000 loan with a 4.25% interest rate paying $200 extra per month will start paying more in principal after 56 months'] = function (test) {
```
Comment (ascott1) [inline test/test.js]: Could you add another test case or two and link to an existing online calculator that demonstrates the same results in a comment on this PR?

File: index.js [hunk 3]
```diff
@@ -55,7 +57,9 @@ var amortizationCalc = function(amount, rate, totalTerm, amortizeTerm, principal
     summedInterest = summedInterest + monthlyIntPaid;
     summedPrincipal = summedPrincipal + monthlyPrincPaid;
     amount = amount - monthlyPrincPaid;
-    i += 1;
+    if(!principalBreakingTerm && monthlyPrincPaid > monthlyIntPaid){
```
Comment (ascott1) [inline index.js]: Same typo as aboce

**Comment by ascott1** [comment]
Hi @avantassel overall this looks like a good change. Thank you, much appreciated! 🎊 

I left a few minor comments. Once those are addressed and @contolini chimes in with a review, we can merge this in. After that we'll bump the version number and publish a new release.

File: index.js (hunk 1 above)
Comment (avantassel) [reply index.js]: This came from a master merge, do you want to change it in master?

https://github.com/cfpb/amortize/pull/14/files#diff-168726dbe96b3ce427e7fedce31bb0bcR25

File: test/test.js (hunk 2 above)
Comment (avantassel) [reply test/test.js]: There aren't any calculators that do this, hence why I added it.  With [bank rate](https://www.bankrate.com/calculators/mortgages/mortgage-calculator.aspx) you can enter extra principal payments and look at the graph then figure it out.

File: index.js (hunk 1 above)
Comment (contolini) [reply index.js]: Good call @avantassel, I'll fix that in master.

Comment (ascott1) [reply index.js]: Ahh yes! Thanks @avantassel and @contolini

**Comment by contolini** [comment]
Thanks @avantassel! I squashed this into https://github.com/cfpb/amortize/commit/b946c36dcfbc77f309fcf3658ef64751a1a146d4 and published it under v1.1.0.

**Pull request closed without merging**